docker.go

```diff
@@ -516,17 +518,63 @@ var _ ContainerProvider = (*DockerProvider)(nil)
 
 // NewDockerProvider creates a Docker provider with the EnvClient
 func NewDockerProvider() (*DockerProvider, error) {
-	client, err := client.NewEnvClient()
+	content := readTCPropsFile()
+	host := grepDockerhost(content)
+
+	opts := []client.Opt{client.FromEnv}
+	if host != "" {
+		opts = append(opts, client.WithHost(host))
+	}
+
+	c, err := client.NewClientWithOpts(opts...)
 	if err != nil {
 		return nil, err
 	}
-	client.NegotiateAPIVersion(context.Background())
+
+	_, err = c.Ping(context.TODO())
+	if err != nil {
+		if host != "" {
+			// fallback to environment
+			c, err = client.NewClientWithOpts(client.FromEnv)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	c.NegotiateAPIVersion(context.Background())
 	p := &DockerProvider{
-		client: client,
+		client: c,
 	}
 	return p, nil
 }
 
+// readTCPropsFile reads from testcontainers properties file, if it exists
+func readTCPropsFile() string {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return ""
+	}
+	tcProp := path.Join(home, ".testcontainers.properties")
+	content, err := os.ReadFile(tcProp)
```

Comment:
We should use a backwards compatible way of reading files OR properly define a support matrix